python/ray/serve/controller.py

```diff
@@ -184,6 +184,8 @@ async def create_endpoint(
             self.endpoint_state.create_endpoint(endpoint, route, methods,
                                                 TrafficPolicy(traffic_dict))
 
+        await self.http_state.ensure_http_route_exists(endpoint, timeout_s=30)
```

Comment:
it'd be better to use the existing `GoalId` mechanism for this - return a goal id, endpoint state performs this check and updates the goal when the futures return, etc. This is likely a temporary solution so it seems ok for now